### PR TITLE
Dp24 resource updates

### LINKED ISSUE #252: Resources review - specify for each module (
### Description of feature

We should avoid using the defaults (process_low, process_high etc). This is as we know the resource requirements of the processes, so should be able to provide these specifically.

### PR DESCRIPTION
**Pull request by DLBPointon**
closes #252 

Updates to resource allocations, mostly from Yumi's brain. Some have been confirmed  by SummaryStats

### CODE REVIEW

**Review by weaglesBio — APPROVED** [review]
Looks okay, but is there any possible consequence of changing the resources for a nextflow defined label (process_single). E.g. overwritten on a upgrade, or assumed by nf-core modules?

File: conf/base.config [hunk 1]
```diff
@@ -91,23 +91,39 @@ process {
...
+    }
+
+    withName: BEDTOOLS_SORT {
+        memory  = { check_max( 1.GB     * Math.ceil( genome_file.countLines() * Math.log(10) * genome_file.countLines())        * task.attempt, 'memory'    ) }
+    }
+
     withName: SELFCOMP_ALIGNMENTBLOCKS {
         cpus    = { check_max( 20       * task.attempt, 'cpus'      ) }
-        memory  = { check_max( 120.GB   * task.attempt, 'memory'    ) }
+        memory  = { check_max( 1.GB     * (  Math.ceil( ( bedfile.size() / 1e+9 ) * 30 ) * Math.ceil( task.attempt * 1 )   ) , 'memory'  ) }
```
Comment (yumisims) [inline conf/base.config:112]: isn't this related with line number in bed file?

Comment (DLBPointon) [reply conf/base.config:112]: These are the notes I made in the meeting, 30*ref for this. Lines in bed was for BEDTOOL-SORT

**Comment by DLBPointon** [comment]
> Looks okay, but is there any possible consequence of changing the resources for a nextflow defined label (process_single). E.g. overwritten on a upgrade, or assumed by nf-core modules?

I didn't realise I changed that, I'll change it back to the default

File: conf/base.config (hunk 1 above)
Comment (DLBPointon) [reply conf/base.config:112]: #252

**Review by yumisims — COMMENTED** [review]
looks good

**Review by weaglesBio — COMMENTED** [review]
Looks fine

**Pull request merged**